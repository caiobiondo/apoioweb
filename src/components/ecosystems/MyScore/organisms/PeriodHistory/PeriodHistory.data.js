import gql from 'graphql-tag';

const buildCycleId = props => {
  let cycleNumber = props.cycleNumber.toString();
  cycleNumber = new Array(2 - cycleNumber.length + 1).join('0') + cycleNumber;
  return props.year + cycleNumber;
};

export const DirectOrdersQuery = gql`
  query DirectOrdersQuery($cycleId: String!) {
    cycleDirectOrders(cycleId: $cycleId) {
      type
      orderNumber
      entryOrderDate
      totalOrderValue
      totalOrderPoints
    }
  }
`;

export const DirectOrdersQueryOptions = {
  options(props) {
    return {
      forceFetch: true,
      variables: {
        cycleId: buildCycleId(props),
      },
    };
  },
  props({ data: { loading, cycleDirectOrders } }) {
    return {
      loadingDirectOrders: loading,
      directOrders: cycleDirectOrders,
    };
  },
};

export const EcommerceOrdersQuery = gql`
  query EcommerceOrdersQuery($cycleId: String!) {
    cycleEcommerceOrders(cycleId: $cycleId) {
      type
      orderNumber
      entryOrderDate
      totalOrderValue
      totalOrderPoints
    }
  }
`;

export const EcommerceOrdersQueryOptions = {
  options(props) {
    return {
      forceFetch: true,
      variables: {
        cycleId: buildCycleId(props),
      },
    };
  },
  props({ data: { loading, cycleEcommerceOrders } }) {
    return {
      loadingEcommerceOrders: loading,
      ecommerceOrders: cycleEcommerceOrders,
    };
  },
};
