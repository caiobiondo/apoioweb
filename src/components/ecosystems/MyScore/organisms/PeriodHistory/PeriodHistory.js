import React from 'react';
import { Icon, Table, Loading } from 'natura-ui';
import { formatCurrency, formatDate } from 'locale/utils';
import { injectIntl, FormattedMessage } from 'react-intl';
import { graphql, compose } from 'react-apollo';
import {
  DirectOrdersQuery,
  DirectOrdersQueryOptions,
  EcommerceOrdersQuery,
  EcommerceOrdersQueryOptions,
} from './PeriodHistory.data';

import {
  Wrapper,
  IconWrapper,
  CloseButton,
  OrderHistoryHeader,
  OrderHistoryTitle,
  OrderHistoryPoints,
  OrderHistoryPoint,
  OrderHistoryPointLabel,
  TableWrapper,
  TableDataWrapper,
} from './PeriodHistory.styles';

const ORDER_ICONS = {
  ecommerce: 'ico_monitor',
  direct: 'ico_marker',
};

const buildTableBody = props => {
  const allOrders = [...props.ecommerceOrders, ...props.directOrders];
  return allOrders.sort((a, b) => {
    if (a.entryOrderDate > b.entryOrderDate) {
      return 1;
    }

    if (a.entryOrderDate < b.entryOrderDate) {
      return -1;
    }

    return 0;
  });
};

const getPointsFrom = orders => {
  if (!orders) {
    return 0;
  }

  return orders.reduce((points, order) => {
    if (!order.totalOrderPoints) {
      return points;
    }

    return points + order.totalOrderPoints;
  }, 0);
};

export const PeriodHistory = props => {
  const { intl } = props;

  if (props.loadingDirectOrders || props.loadingEcommerceOrders) {
    return (
      <Wrapper>
        <Loading />
      </Wrapper>
    );
  }

  const tableData = {
    columns: ['type', 'orderNumber', 'entryOrderDate', 'totalOrderPoints', 'totalOrderValue'],
    style: {
      icon: { width: '10%' },
      orderNumber: { width: '25%' },
      entryOrderDate: { width: '28%' },
      totalOrderPoints: { width: '15%' },
      totalOrderValue: { width: '20%', textAlign: 'right' },
    },
    renderer: {
      type: ({ value }) => (
        <IconWrapper>
          <Icon file={ORDER_ICONS[value]} />
        </IconWrapper>
      ),
      orderNumber: ({ value }) => (
        <TableDataWrapper>
          <FormattedMessage id="order" />&nbsp;
          {value}
        </TableDataWrapper>
      ),
      entryOrderDate: ({ value }) => (
        <TableDataWrapper>
          <FormattedMessage id="date" />&nbsp;
          {formatDate(value, intl)}
        </TableDataWrapper>
      ),
      totalOrderPoints: ({ value }) => `${value} pts`,
      totalOrderValue: ({ value }) => formatCurrency(value, intl),
    },
    body: buildTableBody(props),
  };

  const ecommercePoints = getPointsFrom(props.ecommerceOrders);
  const directPoints = getPointsFrom(props.directOrders);
  const totalPoints = ecommercePoints + directPoints;

  return (
    <Wrapper>
      <CloseButton onClick={props.onClose}>
        <Icon file="ico_times" />
      </CloseButton>
      <OrderHistoryHeader>
        <OrderHistoryTitle>
          <FormattedMessage id="orderHistory" />
        </OrderHistoryTitle>
        <OrderHistoryPoints>
          <OrderHistoryPoint>
            <OrderHistoryPointLabel>
              <span>
                <FormattedMessage id="digital" />:&nbsp;
              </span>
              <Icon file="ico_monitor" />
            </OrderHistoryPointLabel>
            <FormattedMessage id="cyclePoints" values={{ points: ecommercePoints }} />
          </OrderHistoryPoint>

          <OrderHistoryPoint>
            <OrderHistoryPointLabel>
              <span>
                <FormattedMessage id="presential" />:&nbsp;
              </span>
              <Icon file="ico_marker" />
            </OrderHistoryPointLabel>
            <FormattedMessage id="cyclePoints" values={{ points: directPoints }} />
          </OrderHistoryPoint>

          <OrderHistoryPoint>
            <FormattedMessage id="totalInCycle" />:&nbsp;
            <FormattedMessage id="cyclePoints" values={{ points: totalPoints }} />
          </OrderHistoryPoint>
        </OrderHistoryPoints>
      </OrderHistoryHeader>
      <TableWrapper>
        <Table data={tableData} />
      </TableWrapper>
    </Wrapper>
  );
};

const PeriodHistoryIntl = injectIntl(PeriodHistory);

const PeriodHistoryWithData = compose(
  graphql(DirectOrdersQuery, DirectOrdersQueryOptions),
  graphql(EcommerceOrdersQuery, EcommerceOrdersQueryOptions),
)(PeriodHistoryIntl);

export default PeriodHistoryWithData;
