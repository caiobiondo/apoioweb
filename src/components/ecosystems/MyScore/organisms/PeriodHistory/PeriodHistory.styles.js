import styled from 'styled-components';
import { fs12 } from 'styles/typography';
import { gray300 } from 'styles/colors';
import { Paper } from 'natura-ui';
import { Responsive } from '@entria/components';

export const Wrapper = styled(Paper)`
  padding: 50px 100px;
  margin: 10px 0;
  box-shadow: 0 0 25px 0 rgba(0, 0, 0, 0.1);
  position: relative;
  border-radius: 10px !important;

  min-height: 200px;
  > div {
    border-radius: 10px;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    margin: 0 -10%;
    padding: 20px;
  }
`;

export const PageTitle = styled.h1`
  color: #f3971f;
  font-family: Roboto-Light;
  font-size: 21px;
  font-weight: 300;
  line-height: 1.19;
  text-align: left;
`;

export const OrderHistoryHeader = styled.header`
  display: flex;
  justify-content: space-between;
  margin-bottom: 20px;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    flex-direction: column;
  }
`;

export const OrderHistoryTitle = styled.h3`
  display: inline;
  font-size: 22px;
  font-weight: 100;
  margin: 5px;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    align-self: flex-start;
    font-size: 18px;
  }
`;

export const OrderHistoryPoint = styled.li`
  display: inline;
  padding: 7px 15px 5px;
  margin: 10px 0;
  float: left;
  border-left: 1px solid #eee;
  text-transform: capitalize;
  color: #888;
  font-size: ${fs12};

  :first-child {
    border-left: none;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    display: flex;
    margin: 0;
    padding: 10px;
    align-items: center;
  }
`;

export const OrderHistoryPoints = styled.ul`
  display: block;
  border: 1px solid #eee;
  border-radius: 3px;
  margin: -10px 0 20px;
  padding: 0 10px;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    display: flex;
    align-self: flex-start;
    margin: 20px 0 0 0;

    ${OrderHistoryPoint}:first-child {
      padding-left: 0;
    }

    ${OrderHistoryPoint}:last-child {
      padding-right: 0;
    }
  }
`;

export const OrderHistoryPointLabel = styled.div`
  svg {
    fill: ${gray300};
    display: none;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    margin-right: 10px;

    span:first-of-type {
      display: none;
    }

    svg {
      display: inline-block;
      width: 15px;
    }
  }
`;

export const IconWrapper = styled.figure`
  display: block;
  margin: 8px 0 15px;
  width: 25px;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    margin: 0;

    svg {
      width: 15px;
    }
  }
`;

export const CloseButton = styled.a`
  cursor: pointer;
  position: absolute;
  top: 30px;
  right: 27px;
  display: block;

  svg {
    height: 24px;
    width: 24px;
    fill: #000;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    svg {
      height: 12px;
      width: 12px;
    }
  }
`;

export const TableWrapper = styled.div`
  td {
    font-size: 13px !important;
  }
`;

export const TableDataWrapper = styled.div`
  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    > span:first-child {
      display: none;
    }
  }
`;
