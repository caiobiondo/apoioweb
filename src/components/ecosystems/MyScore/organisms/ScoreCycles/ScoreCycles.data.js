import gql from 'graphql-tag';

export const ScoreCyclesQuery = gql`
  query ScoreCyclesQuery($consultantId: Int!, $cycleStart: Int!, $cycleEnd: Int!, $cycle: Int!) {
    scoreCycles(
      consultantId: $consultantId
      cycleStart: $cycleStart
      cycleEnd: $cycleEnd
      cycle: $cycle
    ) {
      totalScore {
        nm_cycle
        vl_score
      }
    }
  }
`;

export const ScoreCyclesQueryOptions = {
  options(props) {
    let cycle = props.growthStatus.cycle;
    let cycleStart = props.growthStatus.periodStartCycle;
    let cycleEnd = props.growthStatus.periodEndCycle;

    if (props.selectedPeriod === 'last') {
      cycle = props.previousPeriod.start;
      cycleStart = 11;
      cycleEnd = 18;
    }

    return {
      forceFetch: true,
      variables: {
        consultantId: props.user.codigo,
        cycleStart,
        cycleEnd,
        cycle,
      },
    };
  },
  props({ data: { loading, scoreCycles } }) {
    return {
      loadingCycles: loading,
      scoreCycles,
    };
  },
};

export const PreviousPeriodQuery = gql`
  query PreviousPeriodQuery($cycle: String!) {
    previousPeriod(cycle: $cycle) {
      start
      year
    }
  }
`;

export const PreviousPeriodQueryOptions = {
  options(props) {
    return {
      forceFetch: true,
      variables: {
        cycle: props.growthStatus.cycle,
      },
    };
  },
  props({ data: { loading, previousPeriod } }) {
    return {
      loadingPreviousPeriod: loading,
      previousPeriod,
    };
  },
};
