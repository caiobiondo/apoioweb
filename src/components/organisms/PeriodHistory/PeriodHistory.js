import React from 'react';
import { Icon, Table } from 'natura-ui';
import { formatCurrency, formatDate } from 'locale/utils';
import { injectIntl, FormattedMessage } from 'react-intl';
import { graphql, compose } from 'react-apollo';
import {
  DirectOrdersQuery,
  DirectOrdersQueryOptions,
  EcommerceOrdersQuery,
  EcommerceOrdersQueryOptions,
} from './PeriodHistory.data';

import {
  Wrapper,
  IconWrapper,
  CloseButton,
  OrderHistoryHeader,
  OrderHistoryTitle,
  OrderHistoryPoints,
  OrderHistoryPoint,
} from './PeriodHistory.styles';

const ORDER_ICONS = {
  ecommerce: 'ico_monitor',
  direct: 'ico_marker',
};

const buildTableBody = props => {
  const allOrders = [...props.ecommerceOrders, ...props.directOrders];
  return allOrders.sort((a, b) => {
    if (a.entryOrderDate > b.entryOrderDate) {
      return 1;
    }

    if (a.entryOrderDate < b.entryOrderDate) {
      return -1;
    }

    return 0;
  });
};

const getPointsFrom = orders => {
  if (!orders) {
    return 0;
  }

  return orders.reduce((points, order) => {
    if (!order.totalOrderPoints) {
      return points;
    }

    return points + order.totalOrderPoints;
  }, 0);
};

const PeriodHistory = props => {
  const { intl } = props;

  if (props.loadingDirectOrders || props.loadingEcommerceOrders) {
    return null;
  }

  const tableData = {
    columns: ['type', 'orderNumber', 'entryOrderDate', 'totalOrderPoints', 'totalOrderValue'],
    style: {
      icon: { width: '10%' },
      orderNumber: { width: '30%' },
      entryOrderDate: { width: '28%' },
      totalOrderPoints: { width: '15%' },
      totalOrderValue: { width: '15%', textAlign: 'right' },
    },
    renderer: {
      type: orderType => (
        <IconWrapper>
          <Icon file={ORDER_ICONS[orderType]} />
        </IconWrapper>
      ),
      orderNumber: value => `Pedido ${value}`,
      entryOrderDate: value => `Data ${formatDate(value, intl)}`,
      totalOrderPoints: value => `${value}pts`,
      totalOrderValue: value => formatCurrency(value, intl),
    },
    body: buildTableBody(props),
  };

  const ecommercePoints = getPointsFrom(props.ecommerceOrders);
  const directPoints = getPointsFrom(props.directOrders);
  const totalPoints = ecommercePoints + directPoints;

  return (
    <Wrapper>
      <CloseButton onClick={props.onClose}>
        <Icon file="ico_times" />
      </CloseButton>
      <OrderHistoryHeader>
        <OrderHistoryTitle>
          <FormattedMessage id="orderHistory" />
        </OrderHistoryTitle>
        <OrderHistoryPoints>
          <OrderHistoryPoint>
            <FormattedMessage id="digital" />:&nbsp;
            <FormattedMessage id="cyclePoints" values={{ points: ecommercePoints }} />
          </OrderHistoryPoint>

          <OrderHistoryPoint>
            <FormattedMessage id="presential" />:&nbsp;
            <FormattedMessage id="cyclePoints" values={{ points: directPoints }} />
          </OrderHistoryPoint>

          <OrderHistoryPoint>
            <FormattedMessage id="totalInCycle" />:&nbsp;
            <FormattedMessage id="cyclePoints" values={{ points: totalPoints }} />
          </OrderHistoryPoint>
        </OrderHistoryPoints>
      </OrderHistoryHeader>
      <Table data={tableData} />
    </Wrapper>
  );
};

const PeriodHistoryIntl = injectIntl(PeriodHistory);

const PeriodHistoryWithData = compose(
  graphql(DirectOrdersQuery, DirectOrdersQueryOptions),
  graphql(EcommerceOrdersQuery, EcommerceOrdersQueryOptions),
)(PeriodHistoryIntl);

export default PeriodHistoryWithData;
