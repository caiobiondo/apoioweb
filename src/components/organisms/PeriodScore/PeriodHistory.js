import React from 'react';
import { Paper, Icon } from 'natura-ui';
import { Wrapper, TableStyle, IconWrapper } from './PeriodHistory.styles';

import {
  Table,
  TableBody,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn,
  TableFooter,
} from 'material-ui/Table';

const tableData = {
  columns: ['icon', 'order', 'date', 'points', 'value'],
  renderer: {
    icon: iconPath => (
      <IconWrapper>
        <Icon file={iconPath} />
      </IconWrapper>
    ),
    order: value => `Pedido ${value}`,
    date: value => `Data ${value}`,
    points: value => `${value}pts`,
    value: value => `R$ ${value}`,
  },
  header: {
    icon: 'Icon',
    order: 'Order',
    date: 'Date',
    points: 'Points',
    value: 'Value',
  },
  body: [
    {
      icon: 'ico_info',
      order: '#361352429',
      date: '24/02/2016',
      points: '10',
      value: '37.34',
    },
    {
      icon: 'ico_info',
      order: '#361352429',
      date: '24/02/2016',
      points: '10',
      value: '37.34',
    },
    {
      icon: 'ico_info',
      order: '#361352429',
      date: '24/02/2016',
      points: '10',
      value: '37.34',
    },
    {
      icon: 'ico_info',
      order: '#361352429',
      date: '24/02/2016',
      points: '10',
      value: '37.34',
    },
  ],
};

const renderCell = (value, key) => {
  if (tableData.renderer && !tableData.renderer[key]) {
    return <TableRowColumn>{value}</TableRowColumn>;
  }
  return <TableRowColumn>{tableData.renderer[key](value)}</TableRowColumn>;
};

const renderTableHeader = () => {
  if (!tableData.header) return null;
  return (
    <TableHeader adjustForCheckbox={false} displaySelectAll={false} enableSelectAll={false}>
      <TableRow style={{ border: 'none' }}>
        {Object.keys(tableData.header).map((key, index) => {
          return <TableHeaderColumn>{tableData.header[key]}</TableHeaderColumn>;
        })}
      </TableRow>
    </TableHeader>
  );
};

const renderTableBody = () => {
  if (!tableData.body) return null;
  return (
    <TableBody displayRowCheckbox={false}>
      {tableData.body.map(row => {
        return (
          <TableRow style={{ border: 'none' }}>
            {Object.keys(row).map((key, index) => {
              return renderCell(row[key], key);
            })}
          </TableRow>
        );
      })}
    </TableBody>
  );
};

const renderTableFooter = () => {
  if (!tableData.footer) return null;
  return (
    <TableFooter adjustForCheckbox={false} displaySelectAll={false} enableSelectAll={false}>
      <TableRow style={{ border: 'none' }}>
        {Object.keys(tableData.header).map((key, index) => {
          return <TableRowColumn>{tableData.header[key]}</TableRowColumn>;
        })}
      </TableRow>
    </TableFooter>
  );
};

const TableExampleSimple = () => (
  <Paper zDepth={2} style={Wrapper}>
    <Table style={TableStyle}>
      {renderTableHeader()}
      {renderTableBody()}
      {renderTableFooter()}
    </Table>
  </Paper>
);

export default TableExampleSimple;
