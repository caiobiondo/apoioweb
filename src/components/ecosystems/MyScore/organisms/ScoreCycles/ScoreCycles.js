import React from 'react';
import {
  ScoreCyclesQuery,
  ScoreCyclesQueryOptions,
  PreviousPeriodQuery,
  PreviousPeriodQueryOptions,
} from './ScoreCycles.data';
import { graphql, compose } from 'react-apollo';
import { injectIntl, FormattedMessage } from 'react-intl';
import withUserData from 'hocs/withUserData/withUserData';
import PeriodHistory from '../PeriodHistory/PeriodHistory';

import { SmallTitle, PeriodTogglerWrapper } from '../MyScore/MyScore.styles';
import { Wrapper } from './ScoreCycles.styles';

import PointsCycleSelection from '../PointsCycleSelection';
import PeriodToggler from '../../molecules/PeriodToggler';

const renderPeriodHistory = props => {
  if (!props.selectedCycleNumber) {
    return null;
  }

  let periodHistoryYear = props.growthStatus.currentPlan.growthPlanYear;

  if (props.selectedPeriod !== 'current') {
    periodHistoryYear = props.previousPeriod.year;
  }

  return (
    <PeriodHistory
      onClose={() => props.cycleSelected(null)}
      year={periodHistoryYear}
      cycleNumber={props.selectedCycleNumber}
    />
  );
};

const ScoreCycles = props => {
  const currentCycleNumber = parseInt(props.growthStatus.parsedCycle.split('/')[0], 10);

  if (props.loadingCycles || props.loadingPreviousPeriod) {
    return null;
  }

  return (
    <Wrapper>
      <SmallTitle>
        <FormattedMessage id="scoreStatementInPeriod" />
      </SmallTitle>

      <PeriodTogglerWrapper>
        <PeriodToggler
          activePeriod={props.selectedPeriod}
          changeSelectedPeriod={props.changeSelectedPeriod}
        />
      </PeriodTogglerWrapper>

      <PointsCycleSelection
        onCycleClick={props.cycleSelected}
        startCycle={props.growthStatus.periodStartCycle}
        endCycle={props.growthStatus.periodEndCycle}
        currentCycleNumber={currentCycleNumber}
        currentLevelColor={props.currentLevel.color}
        scoreCycles={props.scoreCycles.totalScore}
        selectedCycleNumber={props.selectedCycleNumber}
        selectedPeriod={props.selectedPeriod}
      />

      {renderPeriodHistory(props)}
    </Wrapper>
  );
};

export const ScoreCyclesWithIntl = injectIntl(ScoreCycles);

export const ScoreCyclesData = compose(
  graphql(PreviousPeriodQuery, PreviousPeriodQueryOptions),
  graphql(ScoreCyclesQuery, ScoreCyclesQueryOptions),
)(ScoreCyclesWithIntl);

export default withUserData(ScoreCyclesData);
