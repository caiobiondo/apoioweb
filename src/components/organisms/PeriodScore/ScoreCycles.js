import React from 'react';
import { ScoreCyclesQuery, ScoreCyclesQueryOptions } from './PeriodScore.data';
import { graphql } from 'react-apollo';
import { injectIntl, FormattedMessage } from 'react-intl';
import withUserData from 'hocs/withUserData/withUserData';

import { SmallTitle, Wrapper } from './PeriodScore.styles';

import PointsCycleSelection from 'components/organisms/PointsCycleSelection/PointsCycleSelection';

const ScoreCycles = props => {
  const currentCycleNumber = parseInt(props.growthStatus.parsedCycle.split('/')[0], 10);

  if (props.loadingCycles) {
    return null;
  }

  return (
    <Wrapper>
      <SmallTitle>
        <FormattedMessage id="scoreStatementInPeriod" />
      </SmallTitle>

      <PointsCycleSelection
        onCycleClick={props.cycleSelected}
        startCycle={props.growthStatus.periodStartCycle}
        endCycle={props.growthStatus.periodEndCycle}
        currentCycleNumber={currentCycleNumber}
        currentLevelColor={props.currentLevel.color}
        scoreCycles={props.scoreCycles.totalScore}
        selectedCycleNumber={props.selectedCycleNumber}
      />
    </Wrapper>
  );
};

export const PeriodScoreWithIntl = injectIntl(ScoreCycles);

export const PeriodScoreWithScoreCycleData = graphql(ScoreCyclesQuery, ScoreCyclesQueryOptions)(
  PeriodScoreWithIntl,
);

export default withUserData(PeriodScoreWithScoreCycleData);
