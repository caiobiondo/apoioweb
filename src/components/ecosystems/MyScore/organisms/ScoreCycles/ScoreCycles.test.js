import React from 'react';
import { shallow } from 'enzyme';
import { ScoreCycles } from './ScoreCycles';
import PeriodHistory from '../PeriodHistory/PeriodHistory';

const setup = propOverrides => {
  const props = Object.assign(
    {
      loadingCycles: false,
      loadingPreviousPeriod: false,
      selectedCycleNumber: 2,
      previousPeriod: {
        year: '2016',
      },
      growthStatus: {
        currentPlan: {
          levels: [{ levelId: 1, levelSequence: 1 }, { levelId: 2, levelSequence: 2 }],
          growthPlanYear: '2017',
        },
        currentLevelId: 2,
        cycle: '201708',
        parsedCycle: '2017/08',
      },
      currentLevel: {
        color: '128, 128, 128',
        id: 2,
        // text: 'current level name',
        // points: 'levelPointsRangeEnd',
        // sequence: 'levelSequence',
      },
      scoreCycles: [
        /* eslint-disable camelcase */
        { nm_cycle: 1, vl_score: 1012, vl_value: 3587 },
        { nm_cycle: 2, vl_score: 0, vl_value: 0 },
        { nm_cycle: 3, vl_score: 1028, vl_value: 4294 },
        /* eslint-enable camelcase */
      ],
    },
    propOverrides,
  );

  const result = shallow(<ScoreCycles {...props} />);

  return {
    props,
    result,
  };
};

describe('ScoreCycles', () => {
  it('renders score cycles', () => {
    // given
    // when
    const { result } = setup();

    // then
    expect(result).toMatchSnapshot();
  });

  describe('when there is no selected cycle', () => {
    it('does not render period history', () => {
      // given
      const props = {
        selectedCycleNumber: null,
      };

      // when
      const { result } = setup(props);

      // then
      expect(result.find(PeriodHistory).exists()).toBeFalsy();
    });
  });

  describe('when loading cycles', () => {
    it('does not render', () => {
      // given
      const props = {
        loadingCycles: true,
        loadingPreviousPeriod: false,
      };

      // when
      const { result } = setup(props);

      // then
      expect(result.exists()).toBeTruthy();
    });
  });

  describe('when loading previous period', () => {
    it('does not render', () => {
      // given
      const props = {
        loadingCycles: false,
        loadingPreviousPeriod: true,
      };

      // when
      const { result } = setup(props);

      // then
      expect(result.exists()).toBeTruthy();
    });
  });
});
