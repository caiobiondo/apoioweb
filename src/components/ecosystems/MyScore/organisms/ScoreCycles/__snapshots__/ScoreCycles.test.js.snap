// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreCycles renders score cycles 1`] = `
<ScoreCyclesstyles__Wrapper>
  <MyScorestyles__SmallTitle>
    <FormattedMessage
      id="scoreStatementInPeriod"
      values={Object {}}
    />
  </MyScorestyles__SmallTitle>
  <MyScorestyles__PeriodTogglerWrapper>
    <PeriodToggler />
  </MyScorestyles__PeriodTogglerWrapper>
  <PointsCycleSelection
    currentCycleNumber={2017}
    currentLevelColor="128, 128, 128"
    selectedCycleNumber={2}
  />
  <Apollo(Apollo(PeriodHistory))
    cycleNumber={2}
    onClose={[Function]}
    year="2016"
  />
</ScoreCyclesstyles__Wrapper>
`;
